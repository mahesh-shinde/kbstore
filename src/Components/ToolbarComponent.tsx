import React from 'react';

class ToolbarComponent extends React.Component<any, any> {
    constructor(props: any) {
        super(props);

        this.state = {
            IsDeleteButtonEnabled: false
        };
    }

    public render(): JSX.Element {
        return (
            <div className="toolbarContainer" aria-label="toolbar container">
                <div className="search-button" aria-label="toolbar">
                <button onClick={this.OnSearch}><i className="button-icon">&#xe11a;&nbsp;</i>Search</button>
                </div>
                {this.props.IsAdmin &&
                    <div className="edit-buttons">

                        <button onClick={this.AddEditClick}><i className="button-icon">&#xe109;&nbsp;</i>Add</button>
                        {/* <button disabled={!this.state.IsDeleteButtonEnabled} onClick={this.OnEditClick}><i className="button-icon">&#xe109;&nbsp;</i>Edit</button> */}
                        <button disabled={!this.state.IsDeleteButtonEnabled} onClick={this.OnDelete}><i className="button-icon">&#xe10a;&nbsp;</i>Delete</button>
                    </div>
                }
            </div>
        );
    }

    public SetDeleteButtonStats = (status: boolean) =>{
        this.setState({IsDeleteButtonEnabled: status});
    }

    private OnSearch =()=> {
        this.props.OnSearch();
    }

    private OnDelete = () => {
        this.props.OnDelete();
    }

    private AddEditClick = () => {
        this.props.history.push('/AddEdit');
    }

    private OnEditClick = () => {
        this.props.OnEditClick && this.props.OnEditClick();
    }
}

export default ToolbarComponent;
