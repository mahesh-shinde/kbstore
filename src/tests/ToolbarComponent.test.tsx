import * as React from 'react';
import Enzyme, { shallow, ShallowWrapper } from 'enzyme';
import { create, ReactTestRenderer } from 'react-test-renderer';

import ToolbarComponent from '../Components/ToolbarComponent';

let snapshot: ReactTestRenderer;



describe('<toolbar />', () => {
    let wrapper;
    beforeEach(() => {
       
    });

    test('it matches the snapshot', () => {
        const toolbar = <ToolbarComponent />;

        let wrapper = shallow(toolbar);
        snapshot = create(toolbar);

        expect(snapshot.toJSON()).toMatchSnapshot();
    });

    it('it should show toolbar container', () => {
        const toolbar = <ToolbarComponent />;

        let wrapper = shallow(toolbar);

        const container = wrapper.find('.toolbarContainer');

        expect(container.length).toBe(1);
    });

    it('it should show search button when logged in user', () => {
        const toolbar = <ToolbarComponent />;

        let wrapper = shallow(toolbar);

        const buttons = wrapper.find('.button-icon');

        expect(buttons.length).toBe(1);
    });

    it('it should show search, Add and delete button when user role is admin', () => {
        const toolbar = <ToolbarComponent IsAdmin ={true} />;

        let wrapper = shallow(toolbar);

        const buttons = wrapper.find('.button-icon');

        expect(buttons.length).toBe(3);
    });
});