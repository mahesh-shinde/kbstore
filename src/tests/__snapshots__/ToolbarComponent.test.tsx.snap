// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<toolbar /> it matches the snapshot 1`] = `
<div
  aria-label="toolbar container"
  className="toolbarContainer"
>
  <div
    aria-label="toolbar"
    className="search-button"
  >
    <button
      onClick={[Function]}
    >
      <i
        className="button-icon"
      >
         
      </i>
      Search
    </button>
  </div>
</div>
`;
